sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/plan/DeploymentStep.java

```diff
@@ -176,25 +177,26 @@ private void setTaskStatus(Protos.TaskID taskID, Status status) {
         }
     }
 
-    private Status getStatus(Map<Protos.TaskID, TaskStatusPair> tasks) {
+    @VisibleForTesting
+    Status getStatus(Map<Protos.TaskID, TaskStatusPair> tasks) {
         if (tasks.isEmpty()) {
             return Status.PENDING;
         }
 
+        Set<Status> statuses = new HashSet<>();
         for (Map.Entry<Protos.TaskID, TaskStatusPair> entry : tasks.entrySet()) {
             String taskId = entry.getKey().getValue();
             Status status = entry.getValue().getStatus();
             logger.info("TaskId: {} has status: {}", taskId, status);
-            if (!status.equals(Status.COMPLETE)) {
-                // Keep and log current status
-                return super.getStatus();
-            }
+
+            statuses.add(status);
         }
 
-        return Status.COMPLETE;
+        return statuses.stream().min(Comparator.comparing(Status::getOrder)).get();
```

Comment:
I think this behavior which we spoke about offline is fine, but should have significant comment with examples etc.

Reply:
Also, there's a change in behavior where the status of super is never consulted.  The status of super could be lower in order.  Should that ever override the Step's behavior?

Reply:
super.getStatus() is ultimately just the _present_ status of the task.

The change I've made to not rely on the enum having a correct order does re-introduce the fallback.